proxyserver/src/main/java/com/lyft/data/proxyserver/wrapper/MultiReadHttpServletRequest.java

```diff
@@ -42,6 +42,18 @@ public MultiReadHttpServletRequest(HttpServletRequest request) throws IOExceptio
     content = bodyInOutputStream.toByteArray();
   }
 
+  /**
+   * rewrite the body of this request with the given string.
+   *
+   * @param newBody
+   */
+  public void rewriteBody(String newBody) throws IOException {
+    ByteArrayInputStream newBodyReader = new ByteArrayInputStream(newBody.getBytes());
+    ByteArrayOutputStream bodyWriteStream = new ByteArrayOutputStream();
+    copy(newBodyReader, bodyWriteStream);
+    content = bodyWriteStream.toByteArray();
```

Comment:
we need to update content length as well, can you please add a few unit tests?

Reply:
I was in a similar boat trying to rewrite presto queries and had the most success adding a ContentProvider method to ProxyServletImpl  `    @Override
    protected ContentProvider proxyRequestContent(HttpServletRequest request, HttpServletResponse response, Request proxyRequest) throws IOException {
        if (request.getMethod().equals("POST") && request.getRequestURI().startsWith("/v1")) {
            String requestBody = CharStreams.toString(request.getReader());
            String newBody = "";// swap out the content however you like
            log.info("Rewriting " + requestBody + " TO " + newBody.replace("\n", " ").toLowerCase().replace("\r", " ").replaceAll(" +", " ").trim());
            Integer contentLength = newBody.getBytes("UTF-8").length;
            // You have to null it out or you get duplicate Content-Length headers and the new one gets ignored
            proxyRequest.header(HttpHeader.CONTENT_LENGTH.asString(), null);
            proxyRequest.header(HttpHeader.CONTENT_LENGTH.asString(), contentLength.toString());
            return new InputStreamContentProvider(new ByteArrayInputStream(newBody.getBytes()));
        } 
        
        return new InputStreamContentProvider(request.getInputStream());
    }
`